### DISCUSSION TITLE
Sphericity dependent on diameq? Chaining groups?

### DISCUSSION THREAD

**Discussion by TheGermanGeologist** [neperfepx/neper#646] [Q&A] [answered]
I'm wondering if it is possible to make the sphericity dependent on the cell size, for example with an expression like this:
```
"diameq:0.7*normal(1,2)+0.3*normal(2,2),1-sphericity:diameq>1.5?lognormal(0.2,0.02):lognormal(0.6,0.1)"
```
When I try this, the program aborts. Is there any way to achieve this, essentially creating a bimodal distribution where one mode differs both in size and sphericity from the other?

Also, is it possible to somehow "chain" multiple groups? For instance group the cells into two groups based on one property, and then divide one of those two groups into one more like so:
```
-group "diameq>=1?1:2" -group "group==2 && sphericity <=0.7 : 3"
```
(essentially, not using the ternary operator as to not overwrite the first group in the process)

How would one go about this?

If these methods currently are not possible, would there be a straightforward way of implementing this, or is a large restructuring of the code / algorithm necessary to achieve something like this? Feel free to point me to relevant source code files.

Thanks a lot!

**Comment by rquey** [comment, marked as answer]
The first one is not available and not necessarily easy to add, but the relevant code is in `src/neper_t/net_tess/net_tess_opt`.

For the second one, no, `-group`s cannot be chained, but you can use `-group "diameq>=1?1:((sphericity<=0.7)?3:2)"` in the case above. (Be aware that the `diameq` value is considered as absolute in the expression.)
